source/boundary_composition/function.cc

```diff
@@ -36,8 +35,9 @@ namespace aspect
                           const Point<dim> &position,
                           const unsigned int compositional_field) const
     {
-      const Point<dim> point = this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system);
-      return function->value(point, compositional_field);
+      Utilities::NaturalCoordinate<dim> point =
+        this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system);
+      return function->value(point.get_point(), compositional_field);
```

Comment:
Like stated above, you can just use `return function->value(Utilities::convert_array_to_point(point.get_coordinates()), compositional_field);